platform/kube/conversion.go

```diff
@@ -53,11 +55,29 @@ func convertLabels(obj meta_v1.ObjectMeta) model.Labels {
 	return out
 }
 
-func convertPort(port v1.ServicePort) *model.Port {
+// Extracts security option for given port from annotation. If there is no such
+// annotation, or the annotation value is not recognized, returns
+// model.SecurityDefault.
+func extractSecurityOption(port v1.ServicePort, obj meta_v1.ObjectMeta) model.SecurityOption {
+	if obj.Annotations == nil {
+		return model.SecurityDefault
+	}
+	switch obj.Annotations[portSecurityAnnotationKey(int(port.Port))] {
+	case "disable":
+		return model.SecurityDisable
+	case "enable":
+		return model.SecurityEnable
+	default:
+		return model.SecurityDefault
+	}
```

Comment:
Can't you just make the decision to enable/disable auth here? Read the annotation, and then decide to enable/disable based on the global auth policy.

Reply:
Possible, but I'm not sure it's the right place to do. It's easier this way to extend to more options later.